src/main/java/org/candlepin/async/tasks/ConsumerMigration.java

```diff
@@ -0,0 +1,100 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+/**
+ * Class responsible for migration of consumers from anonymous orgs.
+ */
+public class ConsumerMigration {
+
+    private static final Logger log = LoggerFactory.getLogger(ConsumerMigration.class);
+
+    private final OwnerCurator ownerCurator;
+    private final ConsumerCurator consumerCurator;
+    private final IdentityCertificateCurator identityCertificateCurator;
+    private final ContentAccessCertificateCurator contentAccessCertificateCurator;
+    private final int batchSize;
+
+    @Inject
+    public ConsumerMigration(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
+        IdentityCertificateCurator identityCertificateCurator,
+        ContentAccessCertificateCurator contentAccessCertificateCurator,
+        Configuration config) {
+        this.ownerCurator = Objects.requireNonNull(ownerCurator);
+        this.consumerCurator = Objects.requireNonNull(consumerCurator);
+        this.identityCertificateCurator = Objects.requireNonNull(identityCertificateCurator);
+        this.contentAccessCertificateCurator = Objects.requireNonNull(contentAccessCertificateCurator);
+        this.batchSize = config.getInt(ConfigProperties.CONSUMER_MIGRATION_BATCH_SIZE);
+    }
+
+    /**
+     * Migrates all consumers of the given anonymous org to the target org.
+     *
+     * @param originOwnerKey key of the source anonymous org
+     * @param destinationOwnerKey key of the destination org
+     */
+    public void migrate(String originOwnerKey, String destinationOwnerKey) {
+        Owner originOwner = this.ownerCurator.getByKey(originOwnerKey);
+        Owner destinationOwner = this.ownerCurator.getByKey(destinationOwnerKey);
+
+        List<String> consumerIds = this.ownerCurator.getConsumerIds(originOwner).list();
+        for (List<String> consumerIdsBlock : Iterables.partition(consumerIds, this.batchSize)) {
+            try {
+                this.ownerCurator.transactional(args -> {
+                    migrateBatch(consumerIdsBlock, destinationOwner);
+                    return null;
+                }).execute();
+            }
+            catch (Exception e) {
+                log.warn("Failed to migrate a batch of consumers.", e);
+                // Retry later
+            }
+        }
+    }
+
+    private void migrateBatch(List<String> consumerIds, Owner destination) {
+        this.consumerCurator.lockAndLoad(consumerIds);
+
+        List<String> idCerts = this.consumerCurator.getIdentityCertIds(consumerIds);
+        List<String> caCerts = this.consumerCurator.getContentAccessCertIds(consumerIds);
+        this.consumerCurator.unlinkIdCertificates(idCerts);
+        this.consumerCurator.unlinkCaCertificates(caCerts);
+
+        this.identityCertificateCurator.deleteByIds(idCerts);
+        this.contentAccessCertificateCurator.bulkDelete(caCerts);
```

Comment:
I think we can do it in a similar way to the CertificateCLeanupJob here:
use the existing `contentAccessCertificateCurator.deleteByIds` method, then `serialCurator.revokeByIds`, without a need for the new `bulkDelete` method.
And same thing for the identity certs

Reply:
Done.